### DISCUSSION TITLE
Bug: ConvertHgToHpa() never converts when SetOption24 is not set.

### DISCUSSION THREAD

**Discussion by Grumium** [arendst/Tasmota#21844] [Q&A] [answered]
### PROBLEM DESCRIPTION

I believe the intention of this function is to convert raw `mmHg` values to `hPa` if the `SetOption24` flag is **not** set.
Currently, this does not work.

https://github.com/arendst/Tasmota/blob/b9fcfa86c9ce9a6897f09844037e025346f4e923/tasmota/tasmota_support/support.ino#L976-L987

### SOLUTION
`if (!isnan(p) && !Settings->flag.pressure_conversion) {}`

**Comment by arendst** [comment, marked as answer]
Pls, do not try to fix "bugs" in code which is around for many years. Try to think of it as not being a bug as you're the first to notice it (while others just use it as it functions as designed).

**Reply by Grumium** [reply]
Okay, maybe I am misunderstanding the usage of this function.
Since this function has not been used anywhere in a sensor driver so far, I consequently assumed that this could have been overlooked.

I am currently writing a sensor driver in which I would like to use this conversion.
When I publish the sensor value in mmHg to the WebUI it shows the wrong unit. Thus, I assumed that this function serves this purpose of converting mmHg(inHg) to hPa?

**Reply by arendst** [reply]
Tasmota internally uses hPa for pressure. If a user wants to see it in mmHg it should enable SO24. Your driver should not need to call ConvertHgToHpa() as the SO24 functionality is handled by ConvertPressure() which you should always call in your driver to change to  the correct unit a user requested.

If your sensor is only able to provide mmHg then you should convert it to hPa first before using ConvertPressure(). Do not use function ConvertHgToHpa() for this as it's output is user controlled (and that's not what you want at that time in code).

See drivers bmp and qmp6988 for examples.

**Reply by Grumium** [reply]
Okay, since `ConvertPressure()` is used so frequently, I assumed `ConvertHgToHpa()`is just a function for convenience if the sensor output is in mmHg. I now see that this is not the case. I can locally convert the value to hPa first, no problem. Thanks.
